src/arduino.cc/builder/builder_utils/utils.go

```diff
@@ -258,6 +289,24 @@ func nonEmptyString(s string) bool {
 	return s != constants.EMPTY_STRING
 }
 
+func CheckIfRecompileIsAvoidable(corePath string, archiveFile string) bool {
```

Comment:
How does this interplay with referenced cores? E.g. when a platform has a platform.txt but does not contain the actual core, but instead references (e.g.) the arduino core? I believe compilation options could then come from either platform?

In general, I wonder if this can be more thoroughly covered? Ideally, a compiled core would keep a file (like the .d files for each .o file) that shows what files were used to create the .a file, so these could all be checked for modifications, as well as noting the compile options / build properties used to build the core. Even more general would be to log the complete compiler commands executed to build a .a file, and then compare that to the commands that *would be* executed on a subsequent run, and only use them if the commands are exactly equal? This is a bit similar to the caching for include detection that is now used.